File: src/main/java/toast/client/modules/combat/BowSpam.java
package toast.client.modules.combat;

import net.minecraft.item.BowItem;
import net.minecraft.util.Identifier;
import toast.client.event.EventImpl;
import toast.client.event.events.player.EventUpdate;
import toast.client.modules.Module;

public class BowSpam extends Module {

    int ticksLeft = -1;

    public BowSpam() {
        super("BowSpam", "Makes your bow fully automatic.", Category.COMBAT, -1);
        this.settings.addMode("Mode", "No Charge", "No Charge", "Medium Charge", "Full Charge");
    }

    @EventImpl
    public void onUpdate(EventUpdate event) {
        if (mc.player == null || mc.world == null || !(mc.player.getMainHandStack().getItem() instanceof BowItem) || mc.player.getMainHandStack().getItem().getPropertyGetter(new Identifier("pulling")).call(mc.player.getMainHandStack(), mc.world, mc.player) == 0.0F) {
            ticksLeft = -1;
            return;
        }
        if (ticksLeft != -1) {
            ticksLeft--;
        } else {
            if (this.isMode("No Charge")) {
                ticksLeft = 2;
            } else if (this.isMode("Medium Charge")) {
                ticksLeft = 4;
            } else if (this.isMode("Full Charge")) {
                ticksLeft = 20;
            }
        }
        if (ticksLeft == 0) {
            mc.interactionManager.stopUsingItem(mc.player);
            ticksLeft = -1;
        }
    }
}

File: src/main/java/toast/client/modules/combat/BowSpam.kt
package toast.client.modules.combat

import com.google.common.eventbus.Subscribe
import net.minecraft.item.BowItem
import net.minecraft.util.Identifier
import toast.client.events.network.EventSyncedUpdate
import toast.client.modules.Module

class BowSpam : Module("BowSpam", "Makes your bow fully automatic.", Category.COMBAT, -1) {
    var ticksLeft = -1

    @Subscribe
    fun onUpdate(event: EventSyncedUpdate?) {
        if (mc.player == null || mc.world == null || mc.player!!.mainHandStack.item !is BowItem) {
            ticksLeft = -1
            return
        }
        if (mc.player!!.mainHandStack.item.getPropertyGetter(Identifier("pulling"))!!.call(mc.player!!.mainHandStack, mc.world, mc.player) == 0.0f) {
            mc.interactionManager!!.stopUsingItem(mc.player)
            ticksLeft = -1
            return
        }

        if (ticksLeft != -1) ticksLeft--
        else {
            when (mode) {
                "No Charge" -> ticksLeft = 2
                "Medium Charge" -> ticksLeft = 4
                "Full Charge" -> ticksLeft = 20
            }
        }
        if (ticksLeft == 0) {
            mc.interactionManager!!.stopUsingItem(mc.player)
            ticksLeft = -1
        }
    }

    init {
        settings.addMode("Mode", "No Charge", "No Charge", "Medium Charge", "Full Charge")
    }
}